src/main/java/htsjdk/samtools/CRAMContainerStreamWriter.java

```diff
@@ -267,24 +260,24 @@ private static void updateTracks(final List<SAMRecord> samRecords, final Referen
      * @throws IllegalAccessException
      * @throws IOException
      */
-    protected void flushContainer() throws IllegalArgumentException, IllegalAccessException, IOException {
+    protected void flushContainer() throws IllegalArgumentException {
 
-        final byte[] refs;
+        final byte[] referenceBases;
         String refSeqName = null;
-        switch (refSeqIndex) {
-            case Slice.MULTI_REFERENCE:
+        switch (SliceType.bySequenceId(refSeqIndex)) {
```

Comment:
The `refSeqIndex` variable in this class takes on all of the values of SliceType, plus legitimate seq indices, plus the sentinel value `REF_SEQ_INDEX_NOT_INITIALIZED`. I wonder if we can unify those somehow. If not now, then certainly in a future PR..

Reply:
The more I see how this code is evolving, the more I think we need a `ReferenceContext` class that wraps what is currently called `SliceType`, plus a legitimate reference ID, that can be used by `Container`, `Slice` and all of these other classes that need to track that same set of components.

Reply:
My `SliceTyped` idea at 05d46996ced1e85e28d0dc61e04713a3ac92b77d is similar, except inheritance instead of composition.

Reply:
I've added ReferenceContext here, but I'm not thrilled about all the duplicated pass-through methods.  What would you think about a `HasReferenceContext` interface or ABC to interact with a ReferenceContext?

Reply:
See my comment below - I think it would eliminate most of the pass-through methods if we just make the one concrete implementation of `ReferenceContext` be the currency used everywhere, but if not let me know and we can revisit this.